### PR TITLE
Add support for Vector Id columns

### PR DESCRIPTION
**Pull request by bamine**

### CODE REVIEW

**Comment by rom1504** [comment]
LGTM, but will need a lot of rebasing.
In particular:
* will need to read the id column from the new ParquetReader
* add a new ids_path option to build_index function to know where to save

File: autofaiss/external/quantize.py [hunk 1]
```diff
@@ -100,15 +108,15 @@ def build_index(
     print(f"Using {nb_cores} omp threads (processes), consider increasing --nb_cores if you have more")
     faiss.omp_set_num_threads(nb_cores)
 
-    if isinstance(embeddings_path, np.ndarray):
+    if isinstance(embeddings_or_path, np.ndarray):
         tmp_dir_embeddings = tempfile.TemporaryDirectory()
-        np.save(tmp_dir_embeddings.name + "/emb.npy", embeddings_path)
-        embeddings_path = tmp_dir_embeddings.name
+        np.save(os.path.join(tmp_dir_embeddings.name, "emb.npy"), embeddings_or_path)
+        embeddings_or_path = tmp_dir_embeddings.name
```
Comment (rom1504) [inline autofaiss/external/quantize.py:114]: Should be embeddings_path starting from this point in the code: it becomes only a path in both branches

Comment (bamine) [reply autofaiss/external/quantize.py]: Done

**Comment by rom1504** [comment]
Looks pretty good overall. Some minor comments:
* there are some places left to update embeddings_path to embeddings_or_path (at least the readme and the examples in examples/ folder)
* is the speed as fast as before (using the column option) ?

**Pull request merged**